### PR TITLE
fix: always throw error when Promise returns an AxiosError

### PR DESCRIPTION
**Pull request by SVonKarajann**
#### Issue: 
When HTTP response status codes 4XX occur, axiosPromiseResult function doesn't handle them properly by throwing an error

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "trailingComma": "es5",
   "printWidth": 100,
-  "singleQuote": true
+  "singleQuote": true,
+  "endOfLine": "auto"
```
[the remark this answers is not in the document]
Comment (SVonKarajann) [reply .prettierrc.json:5]: on VS code prettier would throw errors unless i use this rule

[the remark this answers is not in the document]
Comment (SVonKarajann) [reply .prettierrc.json:5]: prettier-error

[the remark this answers is not in the document]
Comment (panvourtsis) [reply .prettierrc.json:5]: something else is the problem here, i have it in both VSCode and Webstorm and that option is not the problem

[the remark this answers is not in the document]
Comment (SVonKarajann) [reply .prettierrc.json]: removed

**Review by panvourtsis — APPROVED** [review]

**Pull request merged**